src/imitation/algorithms/dagger.py

```diff
@@ -6,39 +6,86 @@
 policy.
 """
 
+import abc
 import dataclasses
 import logging
 import os
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
 
-import cloudpickle
 import gym
 import numpy as np
-from stable_baselines3.common.policies import ActorCriticPolicy
+import torch as th
+from stable_baselines3.common import utils
 
-from imitation.algorithms.bc import BC
+from imitation.algorithms import bc
 from imitation.data import rollout, types
 from imitation.util import util
 
 
-def linear_beta_schedule(rampdown_rounds: int) -> Callable[[int], float]:
-    """Linearly-decreasing schedule for beta (% of time that user act is used).
+class BetaSchedule(abc.ABC):
+    """
+    Determines the value of beta (% of time that demonstrator action is used) over the
+    progression of training rounds.
+    """
 
-    Args:
-      rampdown_rounds: number of rounds over which to anneal beta. Rounds
-        are assumed to be numbered 0, 1, 2, ….
+    @abc.abstractmethod
+    def __call__(self, round_num: int) -> float:
+        """Gives the value of beta for the current round.
 
-    Returns:
-      schedule: function that takes current round number (0, or 1, or 2, etc.)
-        as input & returns current beta as float.
+        Args:
+            round: the current round number. Rounds are assumed to be numbered 0, 1, 2,
+              etc.
+
+        Returns:
+            beta: the fraction of the time to sample a demonstrator action. Robot
+              actions will be sampled the remainder of the time.
+        """
+
+
+class LinearBetaSchedule(BetaSchedule):
     """
-    assert rampdown_rounds > 0
+    Linearly-decreasing schedule for beta (% of time that demonstrator action is used).
+    """
+
+    def __init__(self, rampdown_rounds: int):
+        """
+        Args:
+            rampdown_rounds: number of rounds over which to anneal beta.
+        """
+        self.rampdown_rounds = rampdown_rounds
 
-    def schedule(i: int) -> float:
-        assert i >= 0
-        return min(1, max(0, (rampdown_rounds - i) / rampdown_rounds))
+    def __call__(self, round_num: int) -> float:
+        """Gives the value of beta for the current round.
```

Comment:
You don't have to copy-paste this docstring from the parent class.

Reply:
Thanks for the tip. I removed the repeated docstring!